ondewo_bpi/intent_max_trigger_handler.py

```diff
@@ -0,0 +1,103 @@
+    Session, QueryParameters
+
+
+class IntentMaxTriggerHandler:
+    intent_with_max_number_triggers_dict = {'Default Fallback Intent': 2, 'Default Exit Intent': 2}
+
+    @classmethod
+    def _get_session(cls, nlu_client: Client, session_id: str) -> Session:
+        get_session_request: GetSessionRequest = GetSessionRequest(session_id=session_id,
+                                                                   session_view=Session.View.VIEW_SPARSE)
+        nlu_session: Session = nlu_client.services.sessions.get_session(request=get_session_request)
+        return nlu_session
+
+    @classmethod
+    def _get_matched_intents(cls, nlu_client: Client, session_id: str) -> List[Dict]:
+        session_dict: Dict = MessageToDict(cls._get_session(nlu_client, session_id))
+        matched_intents: List[Dict] = session_dict['sessionInfo']['matchedIntents']
+        return matched_intents
+
+    @classmethod
+    def _get_intent_display_name_list(cls, nlu_client: Client, session_id: str) -> List[str]:
+        matched_intents: List[Dict] = cls._get_matched_intents(nlu_client, session_id)
+        intent_display_name_list: List[str] = []
+        for matched_intent in matched_intents:
+            intent_display_name_list.append(matched_intent['displayName'])
+        return intent_display_name_list
+
+    @classmethod
+    def _get_intent_display_name_counter(cls, nlu_client: Client, session_id: str) -> Counter:
+        intent_display_name_list: List[str] = cls._get_intent_display_name_list(nlu_client, session_id)
+        return Counter(intent_display_name_list)
+
+    @classmethod
+    def _check_if_intent_reached_number_triggers_max(cls, intent_name: str, nlu_client: Client, session_id: str):
+        max_number_triggers_for_intent: Optional[int] = cls.intent_with_max_number_triggers_dict.get(intent_name)
+        if max_number_triggers_for_intent:
+            intent_display_name_counter: Counter = cls._get_intent_display_name_counter(nlu_client, session_id)
+            current_number_triggers: Optional[int] = intent_display_name_counter.get(intent_name)
+            if current_number_triggers and current_number_triggers == max_number_triggers_for_intent:
+                return True
+        return False
+
+    @classmethod
+    def _get_default_exit_intent_request(cls, session_id: str, language_code: str) -> DetectIntentRequest:
+        context: context_pb2.Context = cls._make_context(session_id)
+
+        nlu_request: DetectIntentRequest = DetectIntentRequest(
+            session=session_id,
+            query_input=QueryInput(
+                text=TextInput(
+                    text="",
+                    language_code=language_code,
+                )
+            ),
+            query_params=QueryParameters(
+                contexts=[context]
+            )
+        )
+        return nlu_request
+
+    @classmethod
+    def _make_context(cls, session_id: str) -> context_pb2.Context:
```

Comment:
Set context for triggering an specific intent.
Let's just rename the function for it to be understandable :)